geometry/eolearn/geometry/transformations.py

```diff
@@ -0,0 +1,340 @@
+
+        self.raster_dtype = raster_dtype
+        self.no_data_value = no_data_value
+        self.write_to_existing = write_to_existing
+        self.rasterio_params = rasterio_params
+
+    @staticmethod
+    def _parse_main_params(vector_input, raster_feature):
+        """ Parsing first 2 task parameters - what vector data will be used and in which raster feature it will be saved
+        """
+        if VectorToRaster._is_geopandas_object(raster_feature):
+            warnings.warn('In the new version of VectorToRaster task order of parameters changed. Parameter for '
+                          'specifying vector data or feature has to be before parameter specifying new raster feature',
+                          DeprecationWarning, stacklevel=3)
+            return VectorToRaster._parse_main_params(raster_feature, vector_input)
+
+        if not VectorToRaster._is_geopandas_object(vector_input):
+            vector_input = VectorToRaster._parse_features(vector_input,
+                                                          allowed_feature_types={FeatureType.VECTOR_TIMELESS})
+        raster_feature = next(iter(
+            VectorToRaster._parse_features(raster_feature, allowed_feature_types=FeatureTypeSet.RASTER_TYPES_3D)
+        ))
+
+        return vector_input, raster_feature
+
+    @staticmethod
+    def _is_geopandas_object(data):
+        """ A frequently used check if object is geopandas `GeoDataFrame` or `GeoSeries`
+        """
+        return isinstance(data, (GeoDataFrame, GeoSeries))
+
+    def _get_vector_data(self, eopatch):
+        """ Provides vector data which will be rasterized
+        """
+        if self._is_geopandas_object(self.vector_input):
+            return self.vector_input
+
+        feature_type, feature_name = next(self.vector_input(eopatch))
+        return eopatch[feature_type][feature_name]
+
+    def _get_rasterization_shapes(self, eopatch):
+        """ Returns a generator of pairs of geometrical shapes and values. In rasterization process each shape will be
+        rasterized to it's corresponding value.
+        If there are no such geometries it will return `None`
+        """
+        vector_data = self._get_vector_data(eopatch)
+
+        if 'init' not in vector_data.crs:
+            raise ValueError('Cannot recognize CRS of vector data')
+        vector_data_crs = CRS(vector_data.crs['init'])
+        if eopatch.bbox.crs is not vector_data_crs:
+            raise ValueError("Vector data and EOPatch should be in the same CRS, found '{}' and '{}'"
+                             "".format(eopatch.bbox.crs, vector_data_crs))
+
+        bbox_poly = eopatch.bbox.geometry
+        vector_data = vector_data[vector_data.geometry.intersects(bbox_poly)].copy(deep=True)
+
+        if vector_data.empty:
+            return None
+
+        if not vector_data.geometry.is_valid.all():
+            warnings.warn('Given vector polygons contain some invalid geometries, they will be fixed', RuntimeWarning)
```

Comment:
Just a question, what is considered as an invalid geometry in this case?

Reply:
This property is from `shapely`: [is_valid docs](https://shapely.readthedocs.io/en/stable/manual.html#object.is_valid). Basically the geometry must not be self-intersecting.